schematic/models/validate_attribute.py

```diff
@@ -378,11 +385,148 @@ def generate_content_error(
             error_row=error_row,
             error_col=attribute_name,
             error_message=error_message,
-            error_val=error_val,
+            error_val=invalid_entry,
         )
 
         return error_list, warning_list
 
+    def _get_rule_attributes(
+        val_rule: str, error_col_name: str, dmge: DataModelGraphExplorer
+    ) -> tuple[[str, str, str]]:
+        """Extract different attributes from the given rule
+        Args:
+            val_rule, str:
+            error_col_name, str
+            dmge, DataModelGraphExplorer Object
+        Returns:
+        """
+        rule_parts = val_rule.split(" ")
+        rule_name = rule_parts[0]
+        specified_level = rule_parts[-1].lower()
+        specified_level = (
+            specified_level if specified_level in ["warning", "error"] else None
+        )
+
+        is_schema_error = rule_name == "schema"
+        col_is_recommended = rule_name == "recommended"
+        col_is_required = dmge.get_node_required(node_display_name=error_col_name)
+        return (
+            rule_parts,
+            rule_name,
+            specified_level,
+            is_schema_error,
+            col_is_recommended,
+            col_is_required,
+        )
+
+    def _get_is_na_allowed(
+        node_display_name: str, dmge: DataModelGraphExplorer
+    ) -> bool:
+        """Determine if NAs are allowed based on the original set of rules
+        Args:
+            node_display_name, str: display name for the current attribure
+            dmge, DataModelGraphExplorer
+        Returns:
+            bool: True, if IsNA is one of the rules, else False
+        """
+        # Get -all- of the specified validation rules for the attribute,
+        validation_rule_list = dmge.get_node_validation_rules(
+            node_display_name=node_display_name
+        )
+
+        # Determine if IsNA is one of the rules, if it is return True, else return False
+        if rule_in_rule_list("IsNA", validation_rule_list):
+            return True
+        else:
+            return False
+
+    def _get_error_value_is_na(
+        error_val,
+        na_allowed: bool = False,
+    ) -> bool:
+        """Determine if the erroring value is NA
+        Args:
+            error_val: erroneous value
+        Returns:
+            bool: Returns True, if the error value is evaluated to be NA, and False if not
+        elif (
+            (error_val is None)
+            or pd.isnull(error_val)
+            or (error_val == "<NA>" and not col_is_required)
+        ):
+
+        """
+        not_applicable_strings = [
+            "not applicable",
+        ]
```

Comment:
Is this meant to be added to at a later time?

Reply:
Im not sure, possibly? I just moved it around, but it was in another function previously. I did a tiny little refactor of some functions.

Reply:
If not line 465 could be
`error_val_is_na = error_val.lower() == "not applicable"`

Reply:
Im going to leave it for now, bc there is actually another ticket involving valid values allowing the string 'None'. We might need to add it here.